exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,618 @@
+
+        network_manager_.set_sni_hostname(mqtt_client_->next_layer().native_handle(), config_.value()._mqtt_broker_address);
+
+        logger_.trace("Starting SSL handshake");
+
+        co_await network_manager_.async_handshake(mqtt_client_->next_layer());
+
+        auto will_topic = topic_formatter(
+            { namespace_element, config_.value()._sparkplug_b_group_id, ndeath, config_.value()._sparkplug_b_edge_node_id });
+
+        // SparkPlugB spec specifies that clean start must be true and Session Expiry Interval must be 0
+        auto connect_packet =
+            async_mqtt::v5::connect_packet{ true,
+                                            std::chrono::seconds(100).count(),
+                                            async_mqtt::allocate_buffer(config_.value()._client_id),
+                                            async_mqtt::will(async_mqtt::allocate_buffer(will_topic),
+                                                             async_mqtt::buffer("DEATH"), async_mqtt::qos::at_least_once),
+                                            async_mqtt::allocate_buffer(config_.value()._mqtt_broker_username),
+                                            async_mqtt::allocate_buffer(config_.value()._mqtt_broker_password),
+                                            { async_mqtt::property::session_expiry_interval{ 0 } } };
+
+        logger_.trace("Sending MQTT connection packet...");
+
+        co_await mqtt_client_->send(connect_packet, asio::use_awaitable);
+
+        logger_.trace("MQTT connection packet sent. Waiting for CONNACK...");
+
+        auto connack_received = co_await mqtt_client_->recv(
+            async_mqtt::filter::match, { async_mqtt::control_packet_type::connack }, asio::use_awaitable);
+
+        auto connack_packet = connack_received.template get<async_mqtt::v5::connack_packet>();
+
+        if (connack_packet.code() != async_mqtt::connect_reason_code::success) {
+          logger_.error("Connection to MQTT broker failed with code: {}",
+                        async_mqtt::connect_reason_code_to_str(connack_packet.code()));
+
+          throw std::runtime_error("Connection to MQTT broker failed");
+        }
+
+        logger_.trace("Received CONNACK. Connection successful.");
+
+        co_return;
+      } catch (std::exception const& exc) {
+        logger_.error("Error while connecting to MQTT broker: {}", exc.what());
+      }
+
+      logger_.trace("Waiting for 10 seconds and then trying to connect again...");
+      co_await asio::co_spawn(mqtt_client_->strand(), mqtt_client_->close(asio::use_awaitable), asio::use_awaitable);
+      co_await asio::steady_timer{ io_ctx_, std::chrono::seconds{ 10 } }.async_wait(asio::use_awaitable);
+    }
+  }
+
+  auto ncmd_listener() -> asio::awaitable<void> {
+    co_await subscribe_to_ncmd_topic();
+
+    while (true) {
+      try {
+        co_await process_ncmd_packet();
+      } catch (std::exception const&) {
+        break;
+      }
+    }
```

Comment:
All this method does is destroy the error information produced inside process_ncmd_packet
why do we not care about this error?

Reply:
I will add a log statement so that the error is used.